src/transport.rs

```diff
@@ -11,50 +11,75 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
+use std::sync::Arc;
+
 use async_trait::async_trait;
 
-use up_rust::{
-    transport::datamodel::UTransport,
-    uprotocol::{UMessage, UStatus, UUri},
-    uuid::builder::UUIDBuilder,
-};
+use up_rust::{UCode, UListener, UMessage, UStatus, UTransport, UUri};
 
 use crate::UPClientMqtt;
 
 #[async_trait]
 impl UTransport for UPClientMqtt {
     async fn send(&self, message: UMessage) -> Result<(), UStatus> {
-        // implementation goes here
-        println!("Sending message: {:?}", message);
+        // validate message
```

Comment:
May be able to make use of the [`UAttributesValidator`](https://github.com/eclipse-uprotocol/up-rust/blob/34a3e79311ce956d55fa07d44830db9b0268f997/src/uattributes/uattributesvalidator.rs#L29) to simplify some of the logic here and in other associated functions of `UPClientMqtt`.

Reply:
Good idea, I have added the validator for the processing of the uAttributes (either in or out) to ensure everything follows the uMessage conventions